### ISSUE TITLE
how to run the test on parametric UMAP

### ISSUE THREAD

**Issue by none0none** [lmcinnes/umap#748]
Hi,
I wonder if I can check this below sample code with you. I have checked whole tutorials and documents of UMAP but I did not find any straight forward remark.
1- I train my Parametric UMAP with one image and the image is clustered (segmentation inside one medical image), then I want to validate it. The question is that how? 

2-Shall I use the same clustered image result with 3 vectors (imagine my n_component=3)?

3- Why the validation is/is not necessary in this parametric UMAP?

4- How I can apply the same clusters on my test data set?

5- I already guess below code for the last question's answer. Is that correct?
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
import umap
import tensorflow as tf
from umap.parametric_umap import ParametricUMAP

from readData import readData
exec(open("./readData.py").read())

mzs1, data_matrix1, imzml_data1 = readData(
    './P3')
mzs3, data_matrix3, imzml_data3 = readData(
    './P2')
mzs2, data_matrix2, imzml_data2 = readData(
    './P1')

(train_images, Y_train) = (data_matrix1, mzs1)
(test_images3, Y_test3) = (data_matrix3, mzs3)
(test_images2, Y_test2) = (data_matrix2, mzs2)

train_images = train_images.reshape((train_images.shape[0], -1))/255.
test_images3 = test_images3.reshape((test_images3.shape[0], -1))/255.
test_images2 = test_images2.reshape((test_images2.shape[0], -1))/255.

validation_images3 = test_images3.reshape((test_images3.shape[0], -1))/255.
validation_images2 = test_images2.reshape((test_images2.shape[0], -1))/255.

print(train_images.shape, 
      test_images2.shape, validation_images2.shape, 
      test_images3.shape, validation_images3.shape)

embedder = ParametricUMAP(
    n_components=3,
    n_training_epochs=1, # dicates how many total training epochs to run
    n_epochs = 50, # dicates how many times edges are trained per 'epoch' to keep consistent with non-parametric UMAP
    parametric_reconstruction= True,
    reconstruction_validation=validation_images2,
#     parametric_reconstruction_loss_fcn = tf.keras.losses.MSE,
    verbose=True
)

embedding = embedder.fit_transform(train_images)
embedder._history.keys()

fig, axs = plt.subplots(ncols=2, figsize=(10,5))
ax = axs[0]
ax.plot(embedder._history['loss'])
ax.set_ylabel('Cross Entropy')
ax.set_xlabel('Epoch')

ax = axs[1]
ax.plot(embedder._history['reconstruction_loss'], label='train')
ax.plot(embedder._history['val_reconstruction_loss'], label='valid')
ax.legend()
ax.set_ylabel('Cross Entropy')
ax.set_xlabel('Epoch')

plt.show()
fig.savefig('./Result/P3P2P1 Cross Entropy.png', dpi=300, bbox_inches='tight')
print("finish Cross Entropy")
embeddingP_NN = np.transpose(embedding)
np.save('./Result/P3P2P1 train.npy', embeddingP_NN)

test_images_recon2 = embedder.transform(test_images2.reshape(len(test_images2),14000))
embeddingP_NN = np.transpose(test_images_recon2)
np.save('./Result/P3P2P1 test2.npy', embeddingP_NN)

test_images_recon3 = embedder.transform(test_images3.reshape(len(test_images3),14000))
embeddingP_NN = np.transpose(test_images_recon3)
np.save('./Result/P3P2P1 test3.npy', embeddingP_NN)

I wish you would add a very simple example in you document which simply show how to apply parametric UMAP using training, validation and test set together in the simple version and not customized one, if you can.

Thanks in advance.
BR,
